### PR TITLE
Task 3: Island & Coordinates (new classes)

### LINKED ISSUE #8: Task 3: Island & Coordinates (new classes)
Napisz klasę Island, która będzie posiadała zmienną Coordinates position_ oraz odpowiedni getter.

Klasa Coordinates ma określać współrzędne na mapie. Również ją napisz. Powinna przyjmować w konstruktorze 2 parametry positionX, positionY oraz operator porównania.

- [x] class Island
- [x] Coordinates _position
- [x] getter 
- [x] class Coordinates
- [x] constructor
- [x] parameters positionX, positionY
- [x] operator ==

### PR DESCRIPTION
**Pull request by Nessy90**

### CODE REVIEW

**Review by kszytko — APPROVED** [review]

**Review by madebykj — APPROVED** [review]

**Review by madebykj — CHANGES_REQUESTED** [review]
Please add file Island.cpp with line #include "Island.hpp"

File: shm/Coordinates.cpp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#include "Coordinates.hpp"
+
+bool Coordinates::operator==(const Coordinates& coordinatesToCheck) const {
+    return _positionX == coordinatesToCheck._positionX &&
+           _positionY == coordinatesToCheck._positionY;
+}
+
+bool Coordinates::operator!=(const Coordinates& coordinatesToCheck) const {
+    return !(this == coordinatesToCheck);
+}
```
Comment (madebykj) [inline shm/Coordinates.cpp:10]: I think that we don't need this at the moment.

**Comment by madebykj** [comment]
From https://coders.school/edu/courses/obiektowosc-1/lesson/projekt-grupowy/ Task 6 description I assume, that Class Coordinates should be part of Class Island. In our situation it is not, so instead of Island* getIsland(const Island::Coordinates& coordinate) in Task 6 I have put Island* getIsland(const Coordinates& coordinate). Otherwise it doesn't work. Correct me if I'm wrong.

**Review by madebykj — APPROVED** [review]

File: shm/Coordinates.cpp [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#include "Coordinates.hpp"
+
+bool Coordinates::operator==(const Coordinates& coordinatesToCheck) const {
+    return _positionX == coordinatesToCheck._positionX &&
+           _positionY == coordinatesToCheck._positionY;
+}
+
+bool Coordinates::operator!=(const Coordinates& coordinatesToCheck) const {
+    return !(this == coordinatesToCheck);
```
Comment (lksobczyk) [inline shm/Coordinates.cpp:9]: return !(*this == coordinatesToCheck);

File: shm/Coordinates.hpp [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    Coordinates(const int positionX,
+                const int positionY)
+        : _positionX(positionX), _positionY(positionY) {}
+
+    bool operator==(const Coordinates& coordinatesToCheck) const;
+    bool operator!=(const Coordinates& coordinatesToCheck) const;
+
+private:
+    const int _positionX;
+    const int _positionY;
```
Comment (lksobczyk) [inline shm/Coordinates.hpp:14]: Why _positionX and _positionY are constant?

**Pull request merged**